path/tree.go

```diff
@@ -139,16 +181,17 @@ func (t *tree) FindAndExtract(path []string, index int) (val interface{}, varMap
 
 func (t *tree) findNode(path []string, index, matchCount int, varMap map[string]interface{}) (found *tree, foundMatchCount int) {
 	if index >= len(path) {
+		// This indicates a complete match. Return MaxInt to ensure it beats others.
 		return t, matchCount + 1
 	}
+
 	name := path[index]
 	if name == "" { // skip empty
 		return t.findNode(path, 1+index, matchCount, varMap)
 	}
 
 	// non-wildcard match
 	if child, ok := t.children[name]; ok {
-		found = child.(*tree)
```

Comment:
Why was this removed?

Reply:
This doesn't have any effect. No test was broken without it. Can you name some example where this is needed?

Reply:
Hmm. I don't think it will affect the outcome, but I believe removing might break the optimization the follows it.

Reply:
I investigated a bit. So with this, it will work like a prefix matching for non-wildcard cases - `/a/b/c/d will return a `/a/b/c` given it is the "best" match. However, in any wildcard cases it breaks in the sense that `/b/c/d` will still return nil even if there is a node `/*`. Adding this same line to the single-wildcard matching condition also won't work for this because it will break other matching behaviors, e.g. `for: a/b/c/d/e/f, want: a/**/c/**/f, got: *`.

I don't see any other optimization this has. `found` is a local value and we don't do any early-termination based on that. Since we are having a separate prefix matching method, I don't think we should keep it.

Reply:
What I'm hearing is inconsistent. You said it didn't have an effect, but after investigation, it does. We must be missing a test. Also, the variable is "local" to the function, but certainly not the block... it could be returned unchanged after this block in the `if foundMatchCount >= len(path)` check and also potentially after that.